orc/src/test/java/org/apache/iceberg/orc/TestOrcDataWriter.java

```diff
@@ -126,4 +135,116 @@ public void testDataWriter() throws IOException {
 
     Assert.assertEquals("Written records should match", records, writtenRecords);
   }
+
+  @Test
+  public void testUsingFileIO() throws IOException {
+    // Show that FileSystem access is not possible for the file we are supplying as the scheme
+    // dummy is not handled
+    ProxyOutputFile outFile = new ProxyOutputFile(Files.localOutput(temp.newFile()));
+    Assertions.assertThatThrownBy(
+            () -> HadoopOutputFile.fromPath(new Path(outFile.location()), new Configuration()))
+        .isInstanceOf(RuntimeIOException.class)
+        .hasMessageStartingWith("Failed to get file system for path: dummy");
+
+    // We are creating the proxy
+    SortOrder sortOrder = SortOrder.builderFor(SCHEMA).withOrderId(10).asc("id").build();
+
+    DataWriter<Record> dataWriter =
+        ORC.writeData(outFile)
+            .schema(SCHEMA)
+            .createWriterFunc(GenericOrcWriter::buildWriter)
+            .overwrite()
+            .withSpec(PartitionSpec.unpartitioned())
```

Comment:
This should be the default, right?

Reply:
Sorry didn't follow this. What do we mean by default?

Before the patch if you gave a Local[Output|Input]File this was converted to FileSystem operation resulting in a LocalFileSystem for handling this.
To ensure that is not taking place, we are mimicking a schema `dummy` that is not handled, so if any FS operations happen normally then it will fail with `Failed to get file system for path: dummy` but if handled via FileIOFS then it should be successful.

I will add this comment to the test to make it clearer.

Reply:
Do we need to add `withSpec` or will it use the unpartitioned spec by default?

Reply:
yeah we need it, without it I get an error that `Cannot create data writer without spec`